### ISSUE TITLE
Detecting DOIs from Zenodo in ORCID import

### ISSUE THREAD

**Issue by wetneb** [dissemin/dissemin#886]
When importing this ORCID profile: https://orcid.org/0000-0003-1066-4559
https://dissem.in/0000-0003-1066-4559
The ORCID profile contains Zenodo DOIs, which should be recognized as open by DOI prefix.

Tasks:
- [ ] check that Zenodo DOI prefix is whitelisted
- [ ] check that the ORCID importer extracts these DOIs correctly
- [ ] check that the extraction of metadata from DOIs work

**Comment by beckstefan** [comment]
This is maybe a general thing with our ORCID importer not recognizing the DOI at all.

I have not checked all publications, but https://dissem.in/p/108330986/open-science-essentials-towards-a-skill-set-amp-showcases/ has DOI `10.5281/zenodo.3702400`. Our Citeproc-parser does not import this DOI, because there is no `container-title`. When looking at the OAI-Record from ORCID-Importer, we see that there's no DOI. So either it is not included in the metadata or not recognized.

Zenodo prefix is in `backend/doiprefixes.py`.

**Comment by wetneb** [comment]
The ORCID importer is supposed to fetch metadata from DOIs in priority when it finds some, but perhaps this has changed.

Note that Zenodo DOIs are minted by Datacite, not Crossref, so fetching them from the Crossref REST API will fail.

**Comment by beckstefan** [comment]
> The ORCID importer is supposed to fetch metadata from DOIs in priority when it finds some, but perhaps this has changed.

No, this has not changed. If there is no DOI or the ingest via DOI fails, the importer falls back to ORCID metadata. In the example above, the latter is the case. What I meant with "not recognizing" is: When processing ORCID metadata to create a OaiRecord, the DOI is not extracted.

> Note that Zenodo DOIs are minted by Datacite, not Crossref, so fetching them from the Crossref REST API will fail.

This is true, but the metadata for those DOI are fetched via resolving the DOI. Our current Citeproc importer can both: CrossRef API and via resolver.

**Comment by wetneb** [comment]
Ah ok!

Perhaps one fix for this would be to add the DOI to the ORCID OaiRecord even if metadata extraction failed (because of a lacking container-title), and apply the DOI prefix whitelist to determine OA status…
